### PR TITLE
JS API: New - Data/Globals. 

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: assets/dev/js/editor/component.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import ComponentBase from 'elementor-api/modules/component-base';
 import Document from './document';
 import * as commands from './commands/';
-import * as internalCommands from './commands/internal';
```
Comment (iNewLegend) [inline assets/dev/js/editor/component.js:4]: Remove

File: assets/dev/js/editor/data/globals/commands/data/typography.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import CommandData from 'elementor-api/modules/command-data';
```
Comment (iNewLegend) [inline assets/dev/js/editor/data/globals/commands/data/typography.js:1]: test

File: assets/dev/js/editor/data/globals/typography/commands/create.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import CreateBase from 'elementor-editor/data/globals/base/create-base';
```
Comment (iNewLegend) [inline assets/dev/js/editor/data/globals/typography/commands/create.js:1]: test

File: assets/dev/js/editor/document/elements/commands/paste-style.js [hunk 4]
```diff
@@ -45,11 +45,25 @@ export class PasteStyle extends CommandHistory {
 			const targetSettings = targetContainer.settings,
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/elements/commands/paste-style.js:45]: test

File: assets/dev/js/editor/document/globals/commands/disable.js [hunk 5]
```diff
@@ -1,19 +1,66 @@
 import DisableEnable from './base/disable-enable';
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/globals/commands/disable.js:1]: test

File: assets/dev/js/editor/document/globals/commands/enable.js [hunk 6]
```diff
@@ -12,6 +12,11 @@ export class Enable extends DisableEnable {
 			container.settings.set( '__globals__', container.globals.toJSON() );
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/globals/commands/enable.js:12]: test

File: assets/dev/js/editor/document/hooks/ui/globals/enable/enqueue-fonts.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import After from 'elementor-api/modules/hooks/ui/after';
+
+export class EnqueueFonts extends After {
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/hooks/ui/globals/enable/enqueue-fonts.js:3]: test

File: assets/dev/js/editor/data/globals/colors/commands/create.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+		let result = false;
+
+		if ( ! controls[ setting ] ) {
+			throw new Error( `Invalid setting: control '${ setting }', not found.` );
+		}
+
+		const id = elementor.helpers.getUniqueID();
+
+		result = $e.data.create( `globals/colors?id=${ id }`, {
```
Comment (matipojo) [inline assets/dev/js/editor/data/globals/colors/commands/create.js:16]: Endpoint or command?
On `$e.data.get` the first parameter is a command.

File: core/common/assets/js/api/core/data/cache.js [hunk 9]
```diff
@@ -181,18 +179,69 @@ export default class Cache {
...
-	 * @param {string} endpoint
+	 * TODO: Create tests for the new logic.
 	 *
-	 * @return {boolean}
+	 * @param {RequestData} requestData
+	 *
+	 * @return {boolean} is deleted
 	 */
-	delete( endpoint ) {
+	delete( requestData ) {
```
Comment (iNewLegend) [inline core/common/assets/js/api/core/data/cache.js:188]: test new logic.

File: assets/dev/js/editor/data/globals/commands/data/index.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import CommandData from 'elementor-api/modules/command-data';
+
+// Alphabetical order
+export { Colors } from './colors';
+export { Typography } from './typography';
+
+// TODO: Remove - Move to into base, Possible to handle using ComponentData.
+export class Index extends CommandData {
+	static getEndpointFormat() {
+		// 'globals/index' => 'globals'.
```
Comment (matipojo) [inline assets/dev/js/editor/data/globals/commands/data/index.js:10]:
```suggestion
		// Format 'globals/index' to 'globals'.
```

File: core/editor/data/globals/endpoints/colors.php [hunk 11]
```diff
@@ -0,0 +1,47 @@
+<?php
```
Comment (iNewLegend) [inline core/editor/data/globals/endpoints/colors.php:1]: test

File: assets/dev/js/editor/data/globals/component.js [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+		super.registerAPI();
+	}
+
+	defaultData() {
+		return this.importCommands( commandsData );
+	}
+
+	onElementorLoaded() {
+		// Add global cache before render.
```
Comment (matipojo) [inline assets/dev/js/editor/data/globals/component.js:35]:
```suggestion
		// Add globals to cache before render.
```

File: core/editor/data/globals/endpoints/index.php [hunk 13]
```diff
@@ -0,0 +1,15 @@
+<?php
```
Comment (iNewLegend) [inline core/editor/data/globals/endpoints/index.php:1]: test?

File: core/editor/data/globals/endpoints/typography.php [hunk 14]
```diff
@@ -0,0 +1,63 @@
+<?php
```
Comment (iNewLegend) [inline core/editor/data/globals/endpoints/typography.php:1]: test

File: assets/dev/js/editor/document/elements/commands/settings.js [hunk 15]
```diff
@@ -121,7 +121,9 @@ export class Settings extends CommandHistoryDebounce {
 				container.settings.set( newSettings );
 			}
 
-			container.render();
+			if ( true !== options.preventDefaultRender ) {
```
Comment (matipojo) [inline assets/dev/js/editor/document/elements/commands/settings.js:124]:
```suggestion
			if ( ! options.preventDefaultRender ) {
```

File: core/kits/assets/js/hooks/ui/document/save/save/delete-globals-cache.js [hunk 16]
```diff
@@ -0,0 +1,23 @@
+import After from 'elementor-api/modules/hooks/data/after';
```
Comment (iNewLegend) [inline core/kits/assets/js/hooks/ui/document/save/save/delete-globals-cache.js:1]: test

File: core/kits/documents/kit.php [hunk 17]
```diff
@@ -92,4 +94,29 @@ protected function get_post_statuses() {
 			'publish' => __( 'Published', 'elementor' ),
 		];
 	}
+
+	public function add_repeater_row( $control_id, $item ) {
+		$meta_key = PageManager::META_KEY;
```
Comment (iNewLegend) [inline core/kits/documents/kit.php:99]: test?

File: core/common/assets/js/api/core/data/storages/prefix-storage.js [hunk 18]
```diff
@@ -0,0 +1,36 @@
+import BaseStorage from 'elementor-api/core/data/storages/base-storage';
+
+export default class PrefixStorage extends BaseStorage {
```
Comment (matipojo) [inline core/common/assets/js/api/core/data/storages/prefix-storage.js:3]:
```suggestion
export default class BasePrefixStorage extends BaseStorage {
```

File: includes/base/controls-stack.php [hunk 19]
```diff
@@ -1026,9 +1026,13 @@ public function get_data( $item = null ) {
 	 * @since 2.0.14
 	 * @access public
 	 */
-	public function get_parsed_dynamic_settings( $setting = null ) {
+	public function get_parsed_dynamic_settings( $setting = null, $settings = null ) {
```
Comment (matipojo) [inline includes/base/controls-stack.php:1029]: Why?

File: includes/controls/groups/base.php [hunk 20]
```diff
@@ -543,7 +543,7 @@ private function start_popover( Controls_Stack $element ) {
 			$control_params = array_replace_recursive( $control_params, $popover_options['settings'] );
 		}
 
-		foreach ( [ 'condition', 'conditions' ] as $key ) {
+		foreach ( [ 'global', 'condition', 'conditions' ] as $key ) {
```
Comment (matipojo) [inline includes/controls/groups/base.php:546]: Can it be removed?

File: tests/qunit/mock/e-data/setup-fetch.js [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+
+		return $e.data.cache.getAsync( requestData );
+	};
+};
+
+export const silence = () => {
+	$e.data.fetch = () => Promise.resolve( {} );
+};
+
+export const free = () => {
```
Comment (matipojo) [inline tests/qunit/mock/e-data/setup-fetch.js:23]: Better names.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/data/globals/commands/data/colors.js [hunk 22]
```diff
@@ -0,0 +1,9 @@
+import CommandData from 'elementor-api/modules/command-data';
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/data/globals/commands/data/colors.js:1]: Handled: https://github.com/elementor/elementor/pull/11622/commits/a2972171d1d8f6d5bd83c2ee29938b557968b05c

File: assets/dev/js/editor/data/globals/commands/data/typography.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/data/globals/commands/data/typography.js:1]: Handled: https://github.com/elementor/elementor/pull/11622/commits/2220150234e320712c2f1e372e1843af6225a984

File: assets/dev/js/editor/data/globals/colors/commands/create.js (hunk 8 above)
Comment (iNewLegend) [reply assets/dev/js/editor/data/globals/colors/commands/create.js:16]: Commands support query as part of the command, the same as backend.

backend:
image

frontend:
image

File: assets/dev/js/editor/data/globals/colors/commands/create.js [hunk 23]
```diff
@@ -0,0 +1,23 @@
+import CreateBase from 'elementor-editor/data/globals/base/create-base';
+
+export class Create extends CreateBase {
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/data/globals/colors/commands/create.js:3]: Handled: https://github.com/elementor/elementor/pull/11622/commits/a2972171d1d8f6d5bd83c2ee29938b557968b05c

File: assets/dev/js/editor/data/globals/typography/commands/create.js (hunk 3 above)
Comment (iNewLegend) [reply assets/dev/js/editor/data/globals/typography/commands/create.js:1]: Handled: https://github.com/elementor/elementor/pull/11622/commits/2220150234e320712c2f1e372e1843af6225a984

File: assets/dev/js/editor/data/globals/commands/data/colors.js (hunk 22 above)
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/data/globals/commands/data/colors.js:1]: Handled: https://github.com/elementor/elementor/pull/11622/commits/2076d19e5784eb8bda5bd56faae962c5b60bcfe9

File: assets/dev/js/editor/data/globals/commands/data/typography.js (hunk 2 above)
Comment (iNewLegend) [reply assets/dev/js/editor/data/globals/commands/data/typography.js:1]: Handled: https://github.com/elementor/elementor/pull/11622/commits/c8523becebcfebc2b285cfb01239ac99bdfdd5fc

File: assets/dev/js/editor/document/elements/commands/paste-style.js (hunk 4 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/elements/commands/paste-style.js:45]: Handled: https://github.com/elementor/elementor/pull/11622/commits/b6840f29da8c7d74941cb2b93248edd5eb31391f

File: core/kits/documents/kit.php (hunk 17 above)
Comment (iNewLegend) [reply core/kits/documents/kit.php:99]: @matipojo

File: assets/dev/js/editor/document/globals/commands/disable.js (hunk 5 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/globals/commands/disable.js:1]: Handled: https://github.com/elementor/elementor/pull/11622/commits/379fb362f8a8b29f053cb092c22320bd7f5000d1

File: assets/dev/js/editor/document/globals/commands/enable.js (hunk 6 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/globals/commands/enable.js:12]: Handled: https://github.com/elementor/elementor/pull/11622/commits/b926b5d19e49bf118cc6adf2f171708babd1a518

File: core/common/assets/js/api/core/data/cache.js (hunk 9 above)
Comment (iNewLegend) [reply core/common/assets/js/api/core/data/cache.js:188]: Handled: https://github.com/elementor/elementor/pull/11622/commits/b902956e46e820afdbd4dbeb107455fe3f850187

File: assets/dev/js/editor/document/commands/base/command-disable-enable.js [hunk 24]
```diff
@@ -0,0 +1,94 @@
+import CommandHistory from './command-history';
+import ElementsSettings from 'elementor-document/elements/commands/settings';
+
+export default class CommandDisableEnable extends CommandHistory {
+	static getName() {
+		elementorModules.ForceMethodImplementation();
```
Comment (kobizz) [inline assets/dev/js/editor/document/commands/base/command-disable-enable.js:6]: Probably it's not related to this PR, but according to our standards, every function name must start with a lower-case character

File: core/editor/data/globals/endpoints/colors.php [hunk 25]
```diff
@@ -0,0 +1,47 @@
...
+		$custom_items = $kit->get_settings_for_display( 'custom_colors' );
+
+		if ( ! $system_items ) {
+			$system_items = [];
+		}
+
+		if ( ! $custom_items ) {
+			$custom_items = [];
+		}
+
```
Comment (kobizz) [inline core/editor/data/globals/endpoints/colors.php:25]: There are already default empty arrays for this controls

File: core/editor/data/globals/endpoints/typography.php [hunk 26]
```diff
@@ -0,0 +1,63 @@
...
+		$custom_items = $kit->get_settings_for_display( 'custom_typography' );
+
+		if ( ! $system_items ) {
+			$system_items = [];
+		}
+
+		if ( ! $custom_items ) {
+			$custom_items = [];
+		}
+
```
Comment (kobizz) [inline core/editor/data/globals/endpoints/typography.php:26]: Same as above

File: core/kits/documents/kit.php [hunk 27]
```diff
@@ -92,4 +94,29 @@ protected function get_post_statuses() {
...
+		$document_settings = $this->get_meta( $meta_key );
+
+		if ( ! $document_settings ) {
+			$document_settings = [];
+		}
+
+		if ( ! isset( $document_settings[ $control_id ] ) ) {
+			$document_settings[ $control_id ] = [];
+		}
+
```
Comment (kobizz) [inline core/kits/documents/kit.php:109]: You can skip the first check. The second one will do them both

**Review by kobizz — CHANGES_REQUESTED** [review]
What about all the empty files?

File: assets/dev/js/editor/document/commands/base/command-disable-enable.js (hunk 24 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/commands/base/command-disable-enable.js:6]: image
Yes + Location is not right. 

need to add it to maintenance and fixes.

File: core/editor/data/globals/endpoints/colors.php (hunk 11 above)
Comment (iNewLegend) [reply core/editor/data/globals/endpoints/colors.php:1]: Handled: https://github.com/elementor/elementor/pull/11622/commits/9922cb4df8c25dadf7381609f5a3d804dbbe86b7

File: core/editor/data/globals/endpoints/typography.php (hunk 14 above)
Comment (iNewLegend) [reply core/editor/data/globals/endpoints/typography.php:1]: Handled: https://github.com/elementor/elementor/pull/11622/commits/386c5bcaad3a9b6135951e9ca0c933cdb73ad4b4

File: core/editor/data/globals/endpoints/index.php (hunk 13 above)
Comment (iNewLegend) [reply core/editor/data/globals/endpoints/index.php:1]: Handled: https://github.com/elementor/elementor/pull/11622/commits/c2fff8b742fffb21d1a7884c8c77c527a7bfee9f

File: core/kits/documents/kit.php (hunk 17 above)
Comment (iNewLegend) [reply core/kits/documents/kit.php:99]: @matipojo, should i test it?

File: assets/dev/js/editor/document/commands/base/command-disable-enable.js (hunk 24 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/commands/base/command-disable-enable.js:6]: @matipojo, maybe add 'task'?

File: core/common/assets/js/api/core/data/storages/prefix-storage.js (hunk 18 above)
Comment (iNewLegend) [reply core/common/assets/js/api/core/data/storages/prefix-storage.js]: Handled: https://github.com/elementor/elementor/pull/11622/commits/3b7a0dfdf645525bb35c2c6fcc289ee85c9df7ff

File: includes/base/controls-stack.php (hunk 19 above)
Comment (iNewLegend) [reply includes/base/controls-stack.php:1029]: I don't remember, should i remove it?

File: includes/controls/groups/base.php (hunk 20 above)
Comment (iNewLegend) [reply includes/controls/groups/base.php:546]: I don't remember, should i remove it?

File: core/editor/data/globals/endpoints/typography.php (hunk 26 above)
Comment (iNewLegend) [reply core/editor/data/globals/endpoints/typography.php]: https://github.com/elementor/elementor/pull/11622/commits/5a93721d550eebb83830727a4a291009e80f83b4

File: core/editor/data/globals/endpoints/colors.php (hunk 25 above)
Comment (iNewLegend) [reply core/editor/data/globals/endpoints/colors.php]: https://github.com/elementor/elementor/pull/11622/commits/5a93721d550eebb83830727a4a291009e80f83b4

File: core/kits/documents/kit.php (hunk 27 above)
Comment (iNewLegend) [reply core/kits/documents/kit.php]: https://github.com/elementor/elementor/pull/11622/commits/5a93721d550eebb83830727a4a291009e80f83b4 fails.

File: assets/dev/js/editor/data/globals/component.js [hunk 28]
```diff
@@ -0,0 +1,33 @@
...
+		super.registerAPI();
+	}
+
+	defaultData() {
+		return this.importCommands( commandsData );
+	}
+
+	onElementorLoaded() {
+		// Add globals to cache before render.
+		$e.data.get( 'globals/index' );
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/data/globals/component.js:31]: Yes.

File: assets/dev/js/editor/data/globals/typography/commands/create.js [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+		if ( controls[ setting ] && controls[ setting ].groupPrefix ) {
+			groupPrefix = controls[ setting ].groupPrefix;
+		} else {
+			throw new Error( `Invalid setting: control '${ setting }', not found.` );
+		}
+
+		if ( groupPrefix ) {
+			Object.entries( controls ).forEach( ( [ key ] ) => {
+				if ( key.includes( groupPrefix ) ) {
+					availableControls[ key.replace( groupPrefix, 'styles_' ) ] = container.settings.get( key );
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply assets/dev/js/editor/data/globals/typography/commands/create.js:21]: Same we do here:
image

do you have any idea how to handle the issue? i will try to find solution.

File: includes/base/controls-stack.php (hunk 19 above)
Comment (matipojo) [reply includes/base/controls-stack.php:1029]: yes.

File: includes/controls/groups/base.php (hunk 20 above)
Comment (matipojo) [reply includes/controls/groups/base.php:546]: We will try after merge with `global-style` panel.

File: core/kits/documents/kit.php (hunk 17 above)
Comment (matipojo) [reply core/kits/documents/kit.php:99]: Yes.

File: assets/dev/js/editor/document/commands/base/command-disable-enable.js (hunk 24 above)
Comment (matipojo) [reply assets/dev/js/editor/document/commands/base/command-disable-enable.js:6]: The TODO is enough.  fix the TODO to `forceMethodImplementation`.

File: assets/dev/js/editor/document/elements/commands/settings.js [hunk 30]
```diff
@@ -121,7 +121,9 @@ export class Settings extends CommandHistoryDebounce {
 				container.settings.set( newSettings );
 			}
 
-			container.render();
+			if ( ! options.preventDefaultRender ) {
```
Comment (matipojo) [inline assets/dev/js/editor/document/elements/commands/settings.js:124]: We still need it?

File: tests/phpunit/elementor/core/editor/data/globals/endpoints/test-colors.php [hunk 31]
```diff
@@ -0,0 +1,46 @@
...
+		$args = [
+			'id' => $id,
+			'color' => 'red',
+			'title' => 'whatever',
+		];
+
+		$result = $this->manager->run_endpoint( $this->get_endpoint( $id ), $args, \WP_REST_Server::CREATABLE );
+
+		// Bug: kit does not updated after save.
+		Plugin::$instance->documents->get( Plugin::$instance->kits_manager->get_active_id(), false );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/editor/data/globals/endpoints/test-colors.php:30]:
```suggestion
		$kit = Plugin::$instance->documents->get( Plugin::$instance->kits_manager->get_active_id(), false );
```

File: tests/phpunit/elementor/core/editor/data/globals/endpoints/test-colors.php [hunk 32]
```diff
@@ -0,0 +1,46 @@
...
+
+		$this->assertEquals( $args, $result );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/editor/data/globals/endpoints/test-colors.php:32]: You should test against the $kit->get_settings()...

File: tests/phpunit/elementor/core/editor/data/globals/endpoints/test-typography.php [hunk 33]
```diff
@@ -0,0 +1,48 @@
...
+			'title' => 'whatever',
+			'value' => [
+				'whatever' => true,
+			],
+		];
+
+		$result = $this->manager->run_endpoint( $this->get_endpoint( $id ), $args, \WP_REST_Server::CREATABLE );
+
+		// Bug: kit does not updated after save.
+		Plugin::$instance->documents->get( Plugin::$instance->kits_manager->get_active_id(), false );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/editor/data/globals/endpoints/test-typography.php:32]: Same as color.

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — DISMISSED** [review]

File: core/kits/documents/kit.php (hunk 27 above)
Comment (kobizz) [reply core/kits/documents/kit.php:109]: What kind of failure?

File: core/common/assets/js/api/core/data/storages/base-prefix-storage.js [hunk 34]
```diff
@@ -0,0 +1,36 @@
+import BaseStorage from 'elementor-api/core/data/storages/base-storage';
```
Comment (kobizz) [inline core/common/assets/js/api/core/data/storages/base-prefix-storage.js:1]: This has to be managed by our existing storage mechanism

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/kits/documents/kit.php (hunk 27 above)
Comment (matipojo) [reply core/kits/documents/kit.php:109]: `array_merge(): Argument #1 is not an array`
https://github.com/elementor/elementor/runs/780041780

File: tests/qunit/mock/e-data/index.js [hunk 35]
```diff
@@ -0,0 +1,8 @@
+export {
+	attachMock,
+	addMock,
+	clearMock,
+	attachCache,
+	restoreFetch,
+	silence,
```
Comment (iNewLegend) [inline tests/qunit/mock/e-data/index.js:7]: ?

File: core/kits/documents/kit.php (hunk 27 above)
Comment (kobizz) [reply core/kits/documents/kit.php:109]: This fail is not related to this comment

File: tests/qunit/core/editor/data/globals/commands/data/colors.spec.js [hunk 36]
```diff
@@ -0,0 +1,39 @@
...
+			eData.addMock( 'create', 'globals/colors' );
+			eData.attachMock();
+		} );
+
+		hooks.after( () => {
+			eData.clearMock();
+		} );
+
+		QUnit.test( 'get', async ( assert ) => {
+			const data = { test: true };
```
Comment (matipojo) [inline tests/qunit/core/editor/data/globals/commands/data/colors.spec.js:16]: Use a real mock.

File: tests/qunit/core/editor/data/globals/commands/data/typography.spec.js [hunk 37]
```diff
@@ -0,0 +1,39 @@
...
+			eData.attachMock();
+		} );
+
+		hooks.after( () => {
+			eData.clearMock();
+			eData.restoreFetch();
+		} );
+
+		QUnit.test( 'get', async ( assert ) => {
+			const data = { test: true };
```
Comment (matipojo) [inline tests/qunit/core/editor/data/globals/commands/data/typography.spec.js:16]: Use a real mock.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/editor-base.js [hunk 38]
```diff
@@ -378,6 +378,8 @@ export default class EditorBase extends Marionette.Application {
 	}
 
 	initPreviewView( document ) {
+        elementor.trigger( 'document:before:preview', document );
```
Comment (kobizz) [inline assets/dev/js/editor/editor-base.js:381]: It seems like a wrong indentation

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/editor-base.js (hunk 38 above)
Comment (iNewLegend) [reply assets/dev/js/editor/editor-base.js:381]: Yes, handled.

File: core/kits/documents/kit.php (hunk 27 above)
Comment (iNewLegend) [reply core/kits/documents/kit.php:109]: > This fail is not related to this comment

i just tried right now, to remove those lines
```
		if ( ! $document_settings ) {
			$document_settings = [];
		}
```
and tests fails.

**Review by matipojo — DISMISSED** [review]

**Review by kobizz — DISMISSED** [review]

**Review by matipojo — DISMISSED** [review]

File: assets/dev/js/editor/elements/views/base.js [hunk 39]
```diff
@@ -442,20 +442,10 @@ BaseElementView = BaseContainer.extend( {
 		} );
 	},
 
-	enqueueFonts() {
+	enqueueIconFonts() {
```
Comment (KingYes) [inline assets/dev/js/editor/elements/views/base.js:445]: BC?

**Review by KingYes — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/elements/views/base.js (hunk 39 above)
Comment (iNewLegend) [reply assets/dev/js/editor/elements/views/base.js]: Handled: https://github.com/elementor/elementor/pull/11622/commits/a0737fe8296b74a26e243a9557a449f254fa5b6a

**Review by matipojo — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**